dask/array/core.py

```diff
@@ -1219,19 +1245,22 @@ def __array__(self, dtype=None, **kwargs):
         return x
 
     def __array_function__(self, func, types, args, kwargs):
-        import dask.array as module
+        if func not in _HANDLED_FUNCTIONS:
+            warnings.warn(
+                "`{}` is not implemented by dask, explicitly "
+                "coerce your array (e.g. with numpy.asarray) "
```

Comment:
I'm somewhat concerned about telling them to coerce their dask arrays to numpy arrays.  Often this will result in them running out of memory. 

I think maybe we should point them towards functions like `map_blocks` instead.

```
"The ... function is not implemented by Dask array. "
"You may want to use the da.map_blocks function "
"or something similar."

Reply:
done. I don't think `map_blocks` is very usable for shape-changing functions though:
```
In [1]: import dask.array as da                                                                

In [2]: a = da.from_array(np.arange(6), chunks=3)                                              

In [3]: a.map_blocks(np.median).compute()                                                      
/Users/rgommers/Code/numpy/numpy/core/fromnumeric.py:3257: RuntimeWarning: Mean of empty slice.
  out=out, **kwargs)
/Users/rgommers/Code/numpy/numpy/core/_methods.py:161: RuntimeWarning: invalid value encountered in double_scalars
  ret = ret.dtype.type(ret / rcount)
---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
...
~/Code/dask/dask/array/core.py in <listcomp>(.0)
   3966 
   3967     while isinstance(arrays, (list, tuple)):
-> 3968         result.append(tuple([shape(deepfirst(a))[dim] for a in arrays]))
   3969         arrays = arrays[0]
   3970         dim += 1

IndexError: tuple index out of range
```

After reading the docs, it should be something like this:
```
a.map_blocks(np.median, chunks=(2,1), axis=[1]).compute()  # still doesn't work ...
```

Reply:
Yeah, it's not a solution for all cases.  It's probably the most common solution though.  We could also just have no recommendation here other than "dask array doesn't implement the foo function".

Reply:
Okay. It says "or similar" so if it's the most common solution, that's still a lot better than nothing imho. So propose to leave as is

Reply:
Your `"or something similar to silence this warning. "` phrasing is also a nice touch

Reply:
Ah, you just beat me to this. Entirely agreed.